lib/Makefile

```diff
@@ -166,6 +169,14 @@ endif
 endif
 CPPFLAGS  += -DZSTD_LEGACY_SUPPORT=$(ZSTD_LEGACY_SUPPORT)
```

Comment:
I see a slight imbalance in the way flags are passed down the chain
between the static of dynamic libraries.
The static library depends on modifications of default flags implicitly updated (or not),
while the dynamic library entirely resets the flag variables during the final recipe.
While it works, I'm afraid such imbalance could impact maintenance (i.e. future modifications).

So here is my recommendation.
- Have a variable `CPPFLAGS_STATICLIB` the same way you already have `CPPFLAGS_DYNLIB`.
    - Initialize it to "empty"
    - Prefer to set it using `=` rather than `:=`. It's a nice convention for the reviewer to pay attention that this value will be updated later on.
- Initialize  `CPPFLAGS_DYNLIB` and `LDFLAGS_DYNLIB` to their _delta_ vs baseline (not the full line), aka:
    - `CPPFLAGS_DYNLIB = -DZSTD_MULTITHREAD`
    - `LDFLAGS_DYNLIB = -pthread`
    - Alternatively, use `CPPFLAGS_MT` to declare starting values as constant (almost the same as currently, but only the delta to baseline, not the full concatenation). In this case, use `:=` to indicate they are supposed to remain unmodified.
- In the recipe for the dynamic library : 
    - `$(LIBZSTD): CPPFLAGS += $(CPPFLAGS_DYNLIB)`   ( note the `+=`)
    - `$(LIBZSTD): LDFLAGS  += $(LDFLAGS_DYNLIB)`
- In the recipe for the static library : 
    - `$(ZSTD_STATLIB): CPPFLAGS += $(CPPFLAGS_STATICLIB)`
- Now, in `%-mt` and `%-nomt`, set these 3 complementary variables to their appropriate values.

The outcome will be similar. But the chain of flag modifications will be symmetric between the static and the dynamic libraries, which makes it easier to follow and update in the future.

Reply:
This suggestion makes a lot of sense, and is definitely easier to reason about.